compiler/circle-partitioner/src/CirclePartitioner.cpp

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Partition.h"
+#include "PartitionPath.h"
+#include "PartitionJson.h"
+#include "Strings.h"
+
+#include <foder/FileLoader.h>
+
+#include <luci/Importer.h>
+#include <luci/Service/Validate.h>
+#include <luci/CircleExporter.h>
+#include <luci/CircleFileExpContract.h>
+#include <luci/Log.h>
+
+#include <arser/arser.h>
+#include <vconone/vconone.h>
+
+#include <iostream>
+#include <string>
+
+using OptionHook = std::function<int(const char **)>;
+
+namespace
+{
+
+const char *opt_prt = "--partition";
+const char *opt_bks = "--backends";
+const char *opt_def = "--default";
```

Comment:
Why `const char *` is used even `arser` receives argument as `std::string`?

Reply:
I didn't want std::string to run in loading time. `const char *` is simple to initalize in module loading.

Reply:
Okay, as it causes any problem, we can go as it is.

FYI, I asked it because of following reasons.

- `char *` and `std::string` is mixed unnecessarily.
- Initialization is occurred twice, while using `std::string` is once.

Reply:
> char * and std::string is mixed unnecessarily.

Could you point out where it is mixed?

> Initialization is occurred twice, while using std::string is once.

Can you explain more about this? I don't know about what TWICE is occured and what ONCE for std::string.

Reply:
Maybe I was confused while I was seeing other files together...
Never mind what I said last :(